scaleway/table_scaleway_registry_image.go

```diff
@@ -0,0 +1,245 @@
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: "Title of the resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		},
+	}
+}
+
+type imageInfo = struct {
+	registry.Image
+	InstanceID   string
+	Region       scw.Region
+	Project      string
+	Organization string
+}
+
+//// LIST FUNCTION
+
+func listRegistryImages(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_registry_Image.listRegistryImages", "region_parsing_error", err)
+		return nil, err
+	}
+
+	// Get Namespace details
+	namespaceData := h.Item.(*registry.Namespace)
+
+	quals := d.KeyColumnQuals
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_registry_Image.listRegistryImages", "connection_error", err)
+		return nil, err
+	}
+
+	// Create SDK objects for Scaleway Registry product
+	registryApi := registry.NewAPI(client)
+
+	req := &registry.ListImagesRequest{
+		Region:      parseRegionData,
+		NamespaceID: &namespaceData.ID,
+		Page:        scw.Int32Ptr(1),
+	}
+	// Additional filters
+	if quals["name"] != nil {
+		req.Name = scw.StringPtr(quals["name"].GetStringValue())
+	}
+
+	// Retrieve the list of Images
+	maxResult := int64(100)
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < maxResult {
+			maxResult = *limit
+		}
+	}
+	req.PageSize = scw.Uint32Ptr(uint32(maxResult))
+
+	var count int
+
+	for {
+		resp, err := registryApi.ListImages(req)
+		if err != nil {
+			plugin.Logger(ctx).Error("scaleway_registry_Image.listRegistryImages", "query_error", err)
+			return nil, err
+		}
+
+		for _, image := range resp.Images {
+			d.StreamListItem(ctx, imageInfo{*image, namespaceData.ID, namespaceData.Region, namespaceData.ProjectID, namespaceData.OrganizationID})
+
+			// Increase the resource count by 1
+			count++
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if resp.TotalCount == uint32(count) {
+			break
+		}
+		req.Page = scw.Int32Ptr(*req.Page + 1)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getRegistryImage(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_registry_Image.getRegistryImage", "region_parsing_error", err)
+		return nil, err
+	}
+
+	if d.KeyColumnQuals["region"].GetStringValue() != region {
+		return nil, nil
+	}
```

Comment:
Do we need this check since we don't pass in `region` as a key column qual?

Reply:
you're right, the region is provided by table registry_namespace